tools/hansel/bio_hansel.xml

```diff
@@ -1,227 +1,269 @@
-                </assert_contents>
-            </output>
-            <output name="match_results.tab">
-                <assert_contents>
-                    <has_text_matching expression="tilename\s+seq\s+is_revcomp\s+contig_id\s+match_index\s+refposition\s+subtype\s+is_pos_tile\s+sample\s+file_path\s+scheme\s+scheme_version\s+qc_status\s+qc_message"/>
-                </assert_contents>
-            </output>
-            <output name="tech_results.tab">
-                <assert_contents>
-                    <has_text_matching expression="sample\s+subtype\s+qc_status\s+qc_message"/>
-                </assert_contents>
-            </output>
-        </test>
-        <test>
-            <param name="type" value="paired"/>
-            <param name="type_of_scheme" value="heidelberg"/>
-            <param name="fastq_input1" value="SRR5646583_SMALL_1.fastq"/>
-            <param name="fastq_input2" value="SRR5646583_SMALL_2.fastq"/>
-            <output name="results.tab">
-                <assert_contents>
-                    <has_text_matching expression="sample\s+scheme\s+scheme_version\s+subtype\s+all_subtypes\s+tiles_matching_subtype\s+are_subtypes_consistent\s+inconsistent_subtypes\s+n_tiles_matching_all\s+n_tiles_matching_all_expected\s+n_tiles_matching_positive\s+n_tiles_matching_positive_expected\s+n_tiles_matching_subtype\s+n_tiles_matching_subtype_expected\s+file_path\s+avg_tile_coverage\s+qc_status\s+qc_message"/>
-                </assert_contents>
-            </output>
-            <output name="match_results.tab">
-                <assert_contents>
-                    <has_text_matching expression="tilename\s+seq\s+freq\s+refposition\s+subtype\s+is_pos_tile\s+is_kmer_freq_okay\s+sample\s+scheme\s+scheme_version\s+qc_status\s+qc_message"/>
-                </assert_contents>
-            </output>
-            <output name="tech_results.tab">
-                <assert_contents>
-                    <has_text_matching expression="sample\s+subtype\s+avg_tile_coverage\s+qc_status\s+qc_message"/>
-                </assert_contents>
-            </output>
-        </test>
-    </tests>
-    <help><![CDATA[
+<tool id="bio_hansel" name="bio_hansel" version="2.0.0">
+  <description>SNV subtyping of genome sequence reads or assemblies</description>
+  <requirements>
+    <requirement type="package" version="2.0.0">bio_hansel</requirement>
+  </requirements>
+  <command detect_errors="exit_code">
+<![CDATA[
+
+#import re
+
+#if $input.type == 'fasta'
+#set $input_files = '"{}"'.format($input.fasta.name)
+  ln -s "$input.fasta" $input_files &&
+#elif $input.type == 'paired'
+#set $_forward_ext = '.fastq.gz' if $re.match(r'.*\.gz$', $input.forward.name) else '.fastq'
```

Comment:
I believe this code is checking the input file name, and if it ends with `.fastq.gz`, it assumes it is in a gzipped format and so links to a file name ending with `.fastq.gz` (so **bio_hansel** can properly interpret the file as being gzipped)?

One issue with this approach is that not all files ending in `.fastq.gz` in Galaxy are gzipped, nor will all gzipped files end in `.fastq.gz`.  I'm wondering if there's a variable you can use to get the actual file `format` (instead of `name`) in the cheetah templating engine?  @Takadonet do you know?

Reply:
Yup @apetkau that's what it does. If there's some place where the file extension/format info is stored, then that would be great and help simplify this checking. (BTW this is the same checking/pre-processing I do with the refseq_masher tool wrapper)

Reply:
I don't know how it would handle that.

Reply:
Could you try something like `$input.forward.is_of_type('fastqsanger.gz')`?  I got this from the the kraken tool https://github.com/galaxyproject/tools-iuc/blob/master/tool_collections/kraken/kraken/kraken.xml#L35 and it appears to work.

I guess you'd also have to add `is_of_type('fastq.gz')` too.

By the way, if you don't add this, and mark files as `fastqsanger.gz` and try to run, I get the following error:

```
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x8b in position 1: invalid start byte
```

That is, without making these changes, it will interpret `fastqsanger.gz` files as regular `fastq` files and then crash.

Reply:
@apetkau I've updated the tool wrapper to use the `is_of_type()` method instead of a regex on the `name` property. Hopefully that fixes the `UnicodeDecodeError`!

Reply:
Awesome, thanks @peterk87. Yep, this fixes the error.  It looks to be working properly with gzipped and non-gzipped fastq files now.